src/features/rewards/mobile/boxMobile/index.tsx

```diff
@@ -99,12 +99,8 @@ export default class BoxMobile extends React.PureComponent<Props, State> {
   }
 
   setView = (view: string, updateHistory: boolean = true) => {
+    const isDetailView = true
```

Comment:
Via this method, `detailView` should always be true. The logic for handling the index `/` view happens elsewhere

The truth table for this should be:

Index page `rewards/`:
`{ detailView: false, settings: false }`
Detail page `rewards/ads`:
`{ detailView: true, settings: false }`
Detail page `rewards/ads-settings`:
`{ detailView: true, settings: true }`

Reply:
@ryanml do you feel this kind of explanation should go in as a code comment? Aim is for everything to be understandable when reading through, and I'm not sure it is without this context.

Reply:
@petemill can update with code commentary, I also pushed another adjustment:

The previous logic would attempt to (incorrectly) toggle the states (so if you were on settings and wanted to go back `setView('settings')` was called to `setState({ settings: !settings})` this didn't work in every case.

I've updated `setView` to explicitly take the view that is to be shown. So in the above case, going back from the settings page would require `setView('detailView')`, and going back from the detail view will require `setView('index')`